server/src/main/java/com/cloud/resourcelimit/ResourceLimitManagerImpl.java

```diff
@@ -1171,6 +1178,27 @@ public ResourceCountCheckTask() {
 
         @Override
         protected void runInContext() {
+            GlobalLock lock = GlobalLock.getInternLock("ResourceCheckTask");
+            try {
+                if (lock.lock(30)) {
+                    ManagementServerHostVO msHost = managementServerHostDao.findOneByLongestRuntime();
+                    if (msHost == null || (msHost.getMsid() != ManagementServerNode.getManagementServerId())) {
+                        s_logger.debug("Skipping the resource counters recalculation task on this management server");
+                        lock.unlock();
```

Comment:
both branches of the if have the unlock() call, maybe bring it outside (as it does not seem to depend on the if.

Reply:
Thanks, fixed

Reply:
Thanks, fixed